internal/pkg/api/metadata_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !integration
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_ParseDownloadRate(t *testing.T) {
+	tests := []struct {
+		name           string
+		raw            json.RawMessage
+		expectedErrMsg string
+		expectedValue  float64
+	}{{
+		name:          "download rate as float",
+		raw:           json.RawMessage(`1000000`),
+		expectedValue: 1000000.00,
+	}, {
+		name:          "download rate as MBps",
+		raw:           json.RawMessage(`"1MBps"`),
+		expectedValue: 1000000.00,
+	}, {
+		name:           "download rate random string",
+		raw:            json.RawMessage(`"toto"`),
```

Comment:
My suggestion for an additional test case would be to include just the suffix that is checked:
```json.RawMessage(`"MBps"`)``` as a test case